### ISSUE TITLE
Document, typos etc.

### ISSUE THREAD

**Issue by sjsepan3** [VBAndCs/sVB-Small-Visual-Basic#12]
Spell-checkers are a blessing and a curse; they can lead you quickly to many errors, but completely skip over correctly-spelled incorrect words that you would have noticed immediately if it had taken you there.
Anyway, here are my observations...

_______________________

EN:

p18:"but things wasn't"-->"but things weren't" [singular/plural ex: negate it to see whether the root words (was/were) fit with the word 'things' - 'things was' versus 'things were']
"I had to build a simple car racing game to get his attention, but
things wasn't that easy for him!"

p19:"y"-->"by"?
"The PDF book that comes with SB makes it hard y focusing on
drawing shapes by using trigonometric functions"

p21:"intension"-->"intention" [roots: 'intense'/'intensity' --> 'intent']
"For about 5 months, I had no intension to do anything further,"

p30:"boarder"-->"border"
" and image brushes to draw the control's boarder,"

p32:"In fact sVB can be too easy for kids."-->"In fact sVB can be easy for kids too."
"In fact sVB can be too easy for kids."
[Normally 'too' is taken to mean 'also'. In this case, the placement affects the meaning; when it modifies 'easy' this way, it takes on a colloqial meaning, as it does with 'too much' or 'much more than it should be'. So 'too easy' becomes 'easier than it should be' instead of the 'also easy' that you intended.]

p34:"tow taps"-->"two tabs"
"It consists of tow taps:"

p41:". but"-->". But" or "; but" [I'm not sure which is more correct]
". but when you reopen the project or switch back to
one of its forms in the opened forms list, you will see the
Global.sb item again."

p49:"thumps"-->"thumbs"
"This border has 8 resizing thumps"

p50:"thumps"-->"thumbs"
"When you hover by mouse over any of these resizing
thumps,"
"thump"-->"thumb"
"You can left-click the thump and drag it along
these directions to resize the control, which will display two
labels that show the control's width and height in
centimeters."

p56:"o"-->"to"
"All you need o worry about is to choose
a unique name for each form and control, which should also
follow the rules of a valid identifier so that sVB can define a
valid variable with this name that you can use in your code.
But, where to set this name?"

p65:"pallet"-->"palette"
3 instances on this page

p69:"tree"-->"three"
"You can select the image from your file system by
clicking the button with the tree dots title to show
the open file dialog."

p74:"summery"-->"summary"
"The popup info includes a summery about the selected
token, and if it is a function or a method, it will show info
about its parameters and return value."

p75:"summery"-->"summary"
"You can add a summery for variables, subroutines, functions,
and dynamic properties by adding one or more comment
lines directly above their definition line, and/or add a
comment at the end of the definition line."

p97:"ream" --> "read"?
"But if you try to ream an item from the array
before creating it, you will get an error"

p100:"jabs" --> "tabs"?
"The fact that the array is accessed by keys, means it is actually a
dictionary, and it can contain jabs, so if you try to read a value of
an item that doesn't exist, the array expression will return an
empty string"
"how to deferential between"-->"How to distinguish between" or "how to tell the difference between" or "how to differentiate between"
"But some existing items may actually contain an
empty string, so how to deferential between the two cases?"
[Deferential is a derivative of root word 'defer' and not applicable here, recommending words related to 'distinct' or 'differ'. Put another way ... I am no English scholar, so I will *defer* to those whose opinions *differ* (or are *distinct*) from mine.]

p124:"van"-->"can"
"You van jump up or down, but be careful, because jumping
up can cause the program to enter into an infinite loop,
unless you provide an exit condition. "

p154:"deferent"-->"different"
"Calling the subroutine name many times is less verbose
than repeating the same code for many times in deferent
places."

p160:
"loose"-->"lose"
"In fact, you should avoid your windows drive (which is mostly the
C:\), and create a folder names "sVB projects" on another drive,
and use it to save you work so you don't loose it when you had
to reformat your windows drive and reinstall it"
"lust"-->"list"
" Now you
will see the name of the project on the project files tab, and the
lust will show the Form1.xaml and global.sb files. You can add a"

p199:"coeds"-->"codes"
"The code 32 represents the space character,
and you can see what next coeds represent!"

p205:" Indeterminate"-->"Indeterminate" (remove space)
"And the " Indeterminate" button changes the checked state of
the "Show TextBox" checkbox to be indeterminate, by
executing this code:"

p212:" Color labels"-->"Color labels" (remove space)
"This property is used in the " Color labels" project in the
samples folder, to create a label for each color in runtime, and
make it show the color name as a tool tip. When you click any
label, its color will be displayed in the preview label."

p235:"especial"-->"special"
"There are 2 especial colors that need your attention:"

p254:"tow"--two
"Note that the above tow RemoveAt calls can be done with
only this single call:"

p286: "punch" --> "bunch"
"So, windows applications are mostly a punch of event handlers
that are waiting for the user to do something."

p434:"to"-->"too"
"Only the last
directory on the path will be deleted, and be careful because all
its child folders and files will be deleted to."

p443:"descries" --> "describes"
"the first Item descries the file type which ill be displayed
in the dialog window in the file types dropdown list."

p487: "regardless it" --> "regardless of whether it"
"Returns True if the current form is loaded, regardless it is hidden
or shown in this moment."

p496:"left" --> "leave"
"not left empty" --> "not left be empty"
"Note that you need to call this method when the user clicks the
button that processes the input data, because he may left some
controls empty and never set the focus on them, so, they never
fired the OnLostFocus event, hence they are not validated yet,
while your logic requires some of them to have data and not left
empty."

p515:"drown" --> "drawn"
"When True, a line segment is automatically drown between
the last point and the start point of the figure to make it a"

p597: missing word(s)?
"The value of the item to"

p604:"tow" --> "two"
"Note the tow RemoveAt calls can be done with only this
single call:"

p646:"un" --> "in"
"The code un the global area will
get the settings from the registry and set them to the two"

p649:"thump" --> "thumb"?
"By default, the progress bar is drawn horizontally, but you can
rotate it (using the rotation thump in the form designer, or using
the Angle property in code)."

p682: "Trabsparent" --> "Transparent"
"Note that you can change the ticks color by using the ForeColor
property, so you can hide the ticks by setting the ForeColor
property to Colors.None or Colors.Trabsparent."

p697:"coeds" --> "codes"
"The code 32 represents the space character,
and you can see what next coeds represent!"

p698:"mare" --> "more"
"You can print many mare characters, but take care that
appending thousands of lines to the textbox will be slow, and
useless as you can't read thousands of lines."

p710: "resulted " --> "resulting"
"An array containing the resulted items."

p727: "tow "--> "two"
"Add tow textboxes on the form, and set their Text property
to: "Welcome to the Small Visual Basic programming language"."

p730:" means, anything" --> " means anything"
"Unlike WriteLine, this will not append a new line character,
which means, anything written to the text window after this call
will be on the same line."

p732:". which" --> 2 sentences, or missing words?
"Note that this is a single timer for all the program. It is easy to
operate with, but sometimes you need to do many things in
different intervals. which need some more work to share this timer
between them, and in some cases this may not be easy (like
having different tasks on deferent forms)."

P740:
"controls" -- > "control"
"sown" --> "down?"
"The following code uses the turtle to draw a flower by
drawing intersected half circles. We controls which parts of
the circle are drawn by setting the pen up and sown:"

p749: "deferent " --> "different"
"This object is deferent than the Timer object, as you can add many
win timers as you want to each form in the project, while there is
only a single Timer object that was suitable to be used in single
Small Basic code files."

**Comment by VBAndCs** [comment]
@sjsepan3 
This is very helpful. 
I made the corrections and updated the pdf on [GitHub](https://github.com/VBAndCs/sVB-Small-Visual-Basic/blob/master/sVB%20Docs.pdf) and [Google Drive](https://drive.google.com/file/d/1pCnB4_7pEBc_wT2u258MmutYCuXwpvjN), and updated the [installer](https://marketplace.visualstudio.com/items?itemName=ModernVBNET.sVBInstaller).
Thanks :100: .

**Comment by sjsepan3** [comment]
Looks good!
BTW, It looks like you made a better choice than to use MY typo below (I had 'be' and 'left' swapped, and your final wording was best.)
"p496:"left" --> "leave"
"not left empty" --> "not left be empty"
"

Finding the guide very helpful!

**Comment by sjsepan3** [comment]
found one today

p405:"ob"-->"on"
"Add a TextBox ob the form, and use this code to allow the
user to change its font properties"

**Comment by VBAndCs** [comment]
@sjsepan3 
Thanks. I had two sources of typos:
- some missing letters because in the middle of the book I got a keyboard with a strange key design, that is too sensitive to dust so some keys didn't respond to some strokes. I got red of that keyboard, and the funny thing the new one I got is too sensitive to any longer key press, so I got many of Thee which passed the auto check but I searched for them manually!
- some swapped letters. I had this issue in both Arabic and English where I type a letter before its preceding one!
But, why on earth does Word accept `ob` as a word?
I fixed it, but and the book will be updated in a next release.
